src/main/java/org/apache/sling/resourceresolver/impl/ResourceResolverImpl.java

```diff
@@ -1117,4 +1142,68 @@ public Resource move(final String srcAbsPath, final String destAbsPath) throws P
         }
         return rsrc;
     }
+
+
+
+    // A very simple tupel implementation which can be used as key in any map
+    public class String3Tupel {
+
+        @Override
+        public int hashCode() {
+            final int prime = 31;
```

Comment:
I think you can use `java.util.Objects.hashCode` here.

Reply:
Eclipse created both `hashCode` and `equals`, so I didn't want to bother with it :-)

Reply:
Well, the maintenance/review cost will be there, so you can ask Eclipse to do the right thing with recent versions :-)

https://stackoverflow.com/a/51775296/112671